compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -15197,6 +15205,7 @@ private STNode parseTupleTypeDescOrExprStartsWithOpenBracket() {
         STNode memberEnd;
         while (!isEndOfListConstructor(peek().kind)) {
             STNode expr = parseTypeDescOrExpr();
+            expr = getTypeDescFromExpr(expr);
```

Comment:
Although we have used this, it is not clean and can cause other issues that are not in the tests eg: for samples like `Foo|[open()]` we still create tuple type with function call inside. Please create an issue to address this.

Reply:
created the issue https://github.com/ballerina-platform/ballerina-lang/issues/33824